tests/resolvers/Mutation/createUserTag.spec.ts

```diff
@@ -0,0 +1,281 @@
+import { OrganizationTagUser } from "../../../src/models";
+import { createRootTagWithOrg, TestUserTagType } from "../../helpers/tags";
+
+let testUser: testUserType;
+let randomUser: testUserType;
+let testOrganization: testOrganizationType;
+let testTag: TestUserTagType;
+let randomTestTag: TestUserTagType;
+let MONGOOSE_INSTANCE: typeof mongoose | null;
+
+beforeAll(async () => {
+  MONGOOSE_INSTANCE = await connect();
+
+  [testUser, testOrganization, testTag] = await createRootTagWithOrg();
+  [, , randomTestTag] = await createRootTagWithOrg();
+  randomUser = await createTestUser();
+});
+
+afterAll(async () => {
+  await disconnect(MONGOOSE_INSTANCE!);
+});
+
+describe("resolvers -> Mutation -> createUserTag", () => {
+  afterEach(() => {
+    vi.doUnmock("../../../src/constants");
+    vi.resetModules();
+    vi.resetAllMocks();
+  });
+
+  it(`throws NotFoundError if no user exists with _id === context.userId`, async () => {
+    const { requestContext } = await import("../../../src/libraries");
+    const spy = vi
+      .spyOn(requestContext, "translate")
+      .mockImplementationOnce((message) => `Translated ${message}`);
+
+    try {
+      const args: MutationCreateUserTagArgs = {
+        input: {
+          organizationId: Types.ObjectId().toString(),
+          name: "TestUserTag",
+        },
+      };
+
+      const context = {
+        userId: Types.ObjectId().toString(),
+      };
+
+      const { createUserTag: createUserTagResolver } = await import(
+        "../../../src/resolvers/Mutation/createUserTag"
+      );
+
+      await createUserTagResolver?.({}, args, context);
+    } catch (error: any) {
+      expect(spy).toBeCalledWith(USER_NOT_FOUND_ERROR.MESSAGE);
+      expect(error.message).toEqual(
+        `Translated ${USER_NOT_FOUND_ERROR.MESSAGE}`
+      );
+    }
+  });
+
+  it(`throws NotFoundError if no organization exists with _id === args.input.organizationId`, async () => {
+    const { requestContext } = await import("../../../src/libraries");
+    const spy = vi
+      .spyOn(requestContext, "translate")
+      .mockImplementationOnce((message) => `Translated ${message}`);
+
+    try {
+      const args: MutationCreateUserTagArgs = {
+        input: {
+          organizationId: Types.ObjectId().toString(),
+          name: "TestUserTag",
+        },
+      };
+
+      const context = {
+        userId: testUser!.id,
+      };
+
+      const { createUserTag: createUserTagResolver } = await import(
+        "../../../src/resolvers/Mutation/createUserTag"
+      );
+
+      await createUserTagResolver?.({}, args, context);
+    } catch (error: any) {
+      expect(spy).toBeCalledWith(ORGANIZATION_NOT_FOUND_ERROR.MESSAGE);
+      expect(error.message).toEqual(
+        `Translated ${ORGANIZATION_NOT_FOUND_ERROR.MESSAGE}`
+      );
+    }
+  });
+
+  it(`throws TAG_NOT_FOUND error if the parent tag is provided (not null) but the same doesn't exist in the organization of the tag being created`, async () => {
```

Comment:
Make it `parentTagId is provided(not null) but no tag exists with _id === args.input.parentTagId`

Reply:
Got it.